docs/sdg_design.md

```diff
@@ -0,0 +1,99 @@
+# Instructlab SDG Library Design
+
+(This is a draft, seeking reviews and architectural consensus on the design of the ilab sdg library)
```

Comment:
Thanks for posting this!

I know previously we were thinking about a different starting point. Could you update this doc to reflect the current starting point of the code pulled out of `instructlab/instructlab` ?  It currently exposes a single `generate_data()` function. Are there steps you would envision taking to evolve from that code to what you envision here? 

Are there any significant challenges you expect to need to overcome with this starting point?

Hopefully what I'm saying makes sense!

Reply:
Thanks @russellb. This absolutely makes sense. I was planning to capture that in issues instead of this document, because there are a lot of small changes that we will have to make on the way. 

But if you suggest, will add those details in here itself

Reply:
I just think it would be helpful to put the proposal in context of what we're working from. Clarify the problems the updated design solves with what we're starting from - that sort of thing